proposals/blittable.md

```diff
@@ -0,0 +1,161 @@
+
+## Detailed design
+[design]: #detailed-design
+
+The language will introduce a new declaration modifier named `blittable` that can be applied to `struct` definitions.  
+
+``` c#
+blittable struct Point 
+{
+    public int X;
+    public int Y;
+}
+```
+
+The compiler will enforce that the fields of such a struct definition fit one of the following categories:
+
+- `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `char`, `float`, `double`, `decimal`, or `bool`.
+- Any `enum` type
+- Any pointer type which points to a `blittable` type
+- Any user defined struct explicitly declared as `blittable`
+
+Any type fitting the definition above, or any element in the list, is considered a valid `blittable` type.
+
+``` c#
+struct User
+{
+    string FirstName;
+    string LastName;
+}
+
+blittable struct ItemData
+{
+    // Error: blittable struct cannot contain field of type User which is not blittable
+    User User;
+    int Id;
+}
+```
+
+Note that a user defined struct must be explicitly declared as `blittable` in order to meet the requirements above.  This is required even if the struct otherwise meets the requirements of `blittable`. 
+
+``` c#
+struct SimplePoint
+{
+    public int X;
+    public int Y;
+}
+
+blittable struct Data
+{
+    // Error: blittable struct cannot contain field of type SimplePoint which is not blittable
+    SimplePoint Point;
+}
+```
+
+The language will also support the ability to constrain generic type parameters to be `blittable` types.  
+
+``` C#
+void M<T>(T p) where T : blittable struct
+{
+    ...
+}
+
+M<Point>(); // Ok
+M<User>();  // Error: Type User does not satisfy the blittable constraint.
+```
+
+One of the primary motivations for `blittable` structs is ease of interop.  As such it's imperative that such a type have it's field ordered in a sequential, or explicit, layout.  An auto layout of fields makes it impossible to reliably interop the data.  
+
+The default today is for a sequential layout so this doesn't represent a substantial change.  However the compiler will make it illegal to mark such types as having an auto layout.  
+
+``` c#
+// Error: A blittable struct may not be marked with an auto layout
+[StructLayout(LayoutKind.Auto)]
+blittable struct LayoutExample 
+{
+    ...
+}
+```
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback of this feature is that it serves a small number of developers: typically low level library authors or frameworks.  Hence it's spending precious language time for a small number of developers. 
+
+Yet these frameworks are are often the basis for the majority of .NET applications out there.  Hence performance / correctness wins at this level can have a riple effect on the .NET ecosystem.  This makes the feature worth considering even with the limited audience. 
```

Comment:
`frameworks are are`
Just a small nitpick from someone who knows basically nothing about what is said in the proposal.